test/llvm-intrinsics/tan.ll

```diff
@@ -0,0 +1,28 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o - | FileCheck %s
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: spirv-val %t.spv
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir"
+
+; CHECK: ExtInst {{[0-9]+}} {{[0-9]+}} {{[0-9]+}} tan 
+
+; Function Attrs: nounwind readnone
+define dso_local spir_func float @foo(float %x) local_unnamed_addr #0 {
+entry:
+  %0 = call float @llvm.tan.f32(float %x)
```

Comment:
nit: that is up to you, I see that we are here a bit inconsistent and some tests have only scalar fp32 versions others have: fp16,fp32,fp64 versions + 1 vector of fp32 version. Up to you to decide if extra testing is needed

Reply:
I have extended the test to cover more cases.